### PR TITLE
use simulacrum everywhere possible

### PR DESCRIPTION
**Pull request by stew**

### CODE REVIEW

File: core/src/main/scala/cats/Foldable.scala [hunk 1]
```diff
@@ -11,7 +11,7 @@ import simulacrum._
  * `foldLeft` methods, which will usually be used by the associationed
  * `Fold[_]` instance.
  *
- * Foldable[F] is implemented in terms of two basic methods:
+b * Foldable[F] is implemented in terms of two basic methods:
```
Comment (non) [inline core/src/main/scala/cats/Foldable.scala]: looks like a typo

**Comment by non** [comment]
Once you fix that typo, :+1: from me.

**Comment by stew** [comment]
but wait, there's more :)

**Comment by stew** [comment]
oops, I have tests fix

**Comment by japgolly** [comment]
Very cool.

**Comment by ceedubs** [comment]
Looking good so far! The build is showing quite a few compile errors in the laws, though.

**Comment by dwijnand** [comment]
Nice work.

**Comment by ceedubs** [comment]
:+1:

**Comment by non** [comment]
great :+1:

**Pull request merged**